.github/workflows/add_new_rust_to_matrix.yml

```diff
@@ -48,16 +48,16 @@ jobs:
       - name: check if the new Rust release is out
         run: |
           set -euxo pipefail
-          UPCOMING="$(yq '.jobs.rust-tests.strategy.matrix.toolchain as $versions |
-              [$versions[] | select(. != "beta" and . != "stable")] as $numerical_versions |
-              [$versions[] | select(. == "beta" or . == "stable")] as $non_numerical_versions |
+          UPCOMING="$(yq ".jobs.rust-tests.strategy.matrix.toolchain as \$versions |
+              [\$versions[] | select(. != \"beta\" and . != \"stable\")] as \$numerical_versions |
+              [\$versions[] | select(. == \"beta\" or . == \"stable\")] as \$non_numerical_versions |
               (
-                  [$numerical_versions[] | sub("\d+\.(\d+)(?:\.\d+)?", "${1}") | to_number]
+                  [\$numerical_versions[] | sub(\"\\d+\\.(\\d+)(?:\\.\\d+)?\", \"\${1}\") | to_number]
                   | sort
                   | .[-1]
-              ) as $max_named_minor |
-              "1.\($max_named_minor + 2).0"
-            ' .github/workflows/ci.yml)"
+              ) as \$max_named_minor |
+              \"1.\\(\$max_named_minor + 2).0\"
+            " .github/workflows/ci.yml)"
```

Comment:
I don't think these edits are accurate..? I mean that the new code is correct but so was the old code -- the only change is that the code is less readable now because it has all the escape sequences and the possibility of footguns from un-escaped `$` values.

Reply:
yeah I think so... should we like ignore some type of `shellcheck` warnings

Reply:
If possible, that'd be great. Otherwise we're making the code worse (less readable + less maintainable) just to satisfy a tool.

This is perhaps even worth a bug report to `shellcheck`, if you're up for it. Might even be a known issue, or something they'd be open to taking a PR for if you're up for that too.

Reply:
The related issue: https://github.com/rhysd/actionlint/issues/26, I think shellcheck cannot do more than that? but we should be able to specify what kind of error we want to suppress in actionlint.